### PR TITLE
Bonnie test fix

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Fixed io:disk:bonnie 

1-added support as bonnie++ available from distro if it use distro version otherwise use upstream
2-ixed pep8 issue in test
Signed-off-by: Praveen K Pandey praveen@linux.vnet.ibm.com

### CODE REVIEW

File: io/disk/bonnie.py [hunk 1]
```diff
@@ -46,21 +46,39 @@ def setUp(self):
         Source:
          http://www.coker.com.au/bonnie++/experimental/bonnie++-1.03e.tgz
         """
+
```
Comment (harish-24) [inline io/disk/bonnie.py]: can you remove this line?

File: io/disk/bonnie.py [hunk 2]
```diff
@@ -46,21 +46,39 @@ def setUp(self):
...
-                            "Bonnie test to run" % package)
+
+        if process.system("which bonnie++", ignore_status=True):
+            smm = SoftwareManager()
+            if not smm.check_installed('bonnie++') \
+                    and not smm.check_installed('bonnie++'):
+                '''Install the package from upstream'''
+                deps = ['gcc', 'make']
+                if distro.detect().name == 'Ubuntu':
+                    deps.extend(['g++'])
```
Comment (harish-24) [inline io/disk/bonnie.py]: Can you extend the `btrfs` check of L62 here?

File: io/disk/bonnie.py [hunk 3]
```diff
@@ -99,15 +108,16 @@ def test(self):
...
+        cmd = ('bonnie++ %s' % " ".join(args))
         if process.system(cmd, shell=True, ignore_status=True):
             self.fail("test failed")
 
     def tearDown(self):
         '''
         Cleanup of disk used to perform this test
         '''
+        if os.listdir(self.source):
+            build.make(self.source, extra_args='clean')
```
Comment (harish-24) [inline io/disk/bonnie.py]: As it will not `uninstall` the binary from system, can you help me why it is needed?

Comment (PraveenPenguin) [reply io/disk/bonnie.py]: aah you are right i thought it will uninstall binary as well

File: io/disk/bonnie.py (hunk 2 above)
Comment (PraveenPenguin) [reply io/disk/bonnie.py]: not get that i think that is suitable if condition can you help me to rectify this

File: io/disk/bonnie.py [hunk 4]
```diff
@@ -46,21 +46,39 @@ def setUp(self):
...
-
-        for package in deps:
-            if not smm.check_installed(package) and not smm.install(package):
-                self.cancel("Fail to install/check %s, which is needed for"
-                            "Bonnie test to run" % package)
+
+        if process.system("which bonnie++", ignore_status=True):
+            smm = SoftwareManager()
+            if not smm.check_installed('bonnie++') \
+                    and not smm.check_installed('bonnie++'):
```
Comment (Naresh-ibm) [inline io/disk/bonnie.py]: I think it should be smm.install('bonnie++').

**Comment by PraveenPenguin** [comment]
closing this PR as new version of this Test raised #784

**Pull request closed without merging**